### PR TITLE
Replace local ConfirmationButton with react-components [WD-5068]

### PR DESCRIPTION
**Pull request by aaryanporwal**
## Done

- [List of work items including drive-bys]

Fixes [list issues/bugs if needed]

## QA

1. Run the LXD-UI:
    - On the demo server via the link posted by @webteam-app below. This is only available for PRs created by collaborators of the repo. Ask @lorumic or @edlerd for access.
    - With a local copy of this branch, run as described [here](../HACKING.md#setting-up-for-development).
2. Perform the following QA steps:
    - [List the steps to QA the new feature(s) or prove that a bug has been resolved]

## Screenshots

[if relevant, include a screenshot or screen capture]

### CODE REVIEW

**Review by edlerd — COMMENTED** [review]
The tests are looking for a modal with a title of "Confirm delete" but that changed to "Confirm deletion" in the upstreamed component. Can you adjust the 3 tests to the new title? Then the test suite should go green.

**Comment by edlerd** [comment]
> The tests are looking for a modal with a title of "Confirm delete" but that changed to "Confirm deletion" in the upstreamed component. Can you adjust the 3 tests to the new title? Then the test suite should go green.

Sorry, this was not 100% correct, in fact this PR changed the title in `DeleteInstanceBtn` to Confirm deletion, where it should have stayed "Confirm delete" and then the tests should not be changed after all to succeed.

**Comment by aaryanporwal** [comment]
> > The tests are looking for a modal with a title of "Confirm delete" but that changed to "Confirm deletion" in the upstreamed component. Can you adjust the 3 tests to the new title? Then the test suite should go green.
> 
> Sorry, this was not 100% correct, in fact this PR changed the title in `DeleteInstanceBtn` to Confirm deletion, where it should have stayed "Confirm delete" and then the tests should not be changed after all to succeed.

Yes I just found that as well. Thanks :)

File: src/pages/instances/actions/StopInstanceBtn.tsx [hunk 1]
```diff
@@ -52,24 +55,29 @@ const StopInstanceBtn: FC<Props> = ({ instance }) => {
...
-        </>
-      }
-      confirmExtra={
-        <ConfirmationForce label="Force stop" force={[isForce, setForce]} />
-      }
-      confirmButtonLabel="Stop"
-      onCancel={() => setForce(false)}
-      onConfirm={handleStop}
-      isDense={true}
-      isDisabled={instance.status === "Stopped"}
```
Comment (edlerd) [inline src/pages/instances/actions/StopInstanceBtn.tsx:71]: The disabled prop is missing in the new version

File: src/pages/profiles/actions/DeleteProfileBtn.tsx [hunk 2]
```diff
@@ -51,23 +55,30 @@ const DeleteProfileBtn: FC<Props> = ({
...
-      toggleCaption={isDeleteIcon ? undefined : "Delete profile"}
-      confirmMessage={
-        <>
-          This will permanently delete profile <ItemName item={profile} bold />.
-          {"\n"}This action cannot be undone, and can result in data loss.
-        </>
-      }
-      confirmButtonLabel="Delete"
-      onConfirm={handleDelete}
-      isDisabled={isDefaultProfile || !featuresProfiles}
```
Comment (edlerd) [inline src/pages/profiles/actions/DeleteProfileBtn.tsx:67]: This disabled prop is also missing

File: src/pages/projects/actions/DeleteProjectBtn.tsx [hunk 3]
```diff
@@ -56,23 +60,30 @@ const DeleteProjectBtn: FC<Props> = ({ project }) => {
...
-      toggleCaption={isDeleteIcon ? undefined : "Delete project"}
-      confirmMessage={
-        <>
-          This will permanently delete project <ItemName item={project} bold />.
-          {"\n"}This action cannot be undone, and can result in data loss.
-        </>
-      }
-      confirmButtonLabel="Delete"
-      onConfirm={handleDelete}
-      isDisabled={isDefaultProject || !isEmpty}
```
Comment (edlerd) [inline src/pages/projects/actions/DeleteProjectBtn.tsx:72]: This disabled should be kept.

**Review by edlerd — COMMENTED** [review]
Please have a look at the disabled delete buttons, some conditions have been lost, there might be more than I commented on.

**Review by edlerd — APPROVED** [review]
LGTM

**Pull request merged**